pkg/cmd/show.go

```diff
@@ -57,48 +50,125 @@ type Unkown struct {
 	Status            interface{} `json:"status"`
 }
 
-func showApplication(ctx context.Context, c client.Client, cmd *cobra.Command, env *types.EnvMeta, appName string) error {
-	var (
-		application corev1alpha2.ApplicationConfiguration
-	)
-	namespace := env.Namespace
-
-	if err := c.Get(ctx, client.ObjectKey{Name: appName, Namespace: namespace}, &application); err != nil {
-		return fmt.Errorf("Fetch application with Err: %s", err)
-	}
-
-	if len(application.Spec.Components) == 0 {
-		cmd.Println("About:")
-		cmd.Printf("  Appset: %s", appName)
-		cmd.Printf("  ENV: %s", namespace)
-		return nil
-	}
+func showApplication(cmd *cobra.Command, env *types.EnvMeta, appName string) error {
 
-	// current only support one component
-	componentName := application.Spec.Components[0].ComponentName
-	component, err := cmdutil.GetComponent(ctx, c, componentName, namespace)
+	app, err := application.Load(env.Name, appName)
 	if err != nil {
-		return fmt.Errorf("Fetch component %s with Err: %s", componentName, err)
+		return err
 	}
-	if component.Labels == nil {
-		return fmt.Errorf("Can't get workloadDef, please check component %s label \"%s\" is correct.",
-			componentName, types.ComponentWorkloadDefLabel)
+
+	cmd.Printf("About:\n\n")
+	table := uitable.New()
```

Comment:
It would be better to only get application detailed information in `showApplication()` and then print it in `uitable`. In this way, the data can reuse by api server. Of course, I can do the job later.